app/src/main/java/com/example/myapplication/ThirdFragment.java

```diff
@@ -36,35 +43,150 @@ public class ThirdFragment extends Fragment {
 
 
     private FragmentThirdBinding binding;
+    ViewGroup v = null;
 
 
     @Override
     public View onCreateView(
             LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState
     ){
         binding = FragmentThirdBinding.inflate(inflater, container, false);
+        v = container;
+        ColorStateList purple = ColorStateList.valueOf(Color.parseColor(("#6750A3")));
```

Comment:
Since you are defining the same variables `blue` and `purple` in this method (onCreateView) and on another one (onViewCreated), you can make it a field in the class, and reuse it.

Reply:
can you show an example of this i don't really understand

Reply:
If you move the two lines below outside the `onCreateView` method (for example, right below line 45 `private FragmentThirdBinding binding;`):

    private FragmentThirdBinding binding;
    private static ColorStateList purple = ColorStateList.valueOf(Color.parseColor("#6750A3"));
    private static ColorStateList blue = ColorStateList.valueOf(Color.parseColor("#73C2F0"));

Then you can reference `blue` and `purple` in both onCreateView and onViewCreated without having to define them again.

The advantage of that is that if you ever want to change the colors, there will be a single place to do that, instead of you having to search for them in multiple places.

Reply:
ok i got it thank you